pandas/tests/reshape/test_join.py

```diff
@@ -550,6 +550,13 @@ def test_join_mixed_non_unique_index(self):
                              index=[1, 2, 2, 'a'])
         tm.assert_frame_equal(result, expected)
 
+    def test_join_non_unique_period_index(self):
+        per_index = pd.period_range('2016-01-01', periods=16, freq='M')
+        per_df = DataFrame([i for i in range(len(per_index))],
+                           index=per_index, columns=['pnum'])
+        df2 = concat([per_df, per_df])
+        per_df.join(df2, how='outer', rsuffix='_df2')
```

Comment:
result = 

and provide an expected result; compare using ``tm.assert_frame_equal``